agents/ibootbar/ibootbar.py

```diff
@@ -341,29 +200,67 @@ def __init__(self, agent, address, port=161, version=2):
+
         **Process** - Fetch values from the ibootbar via SNMP.
+
+        Notes:
+            The most recent data collected is stored in session.data in the
+            structure::
+                >>> response.session['data']
+                {"fields":
+                    {ibootbar:
+                        {outletStatus_0: {"status": 1, "name": Outlet-1, "description": "on"},
+                         outletStatus_1: {"status": 1, "name": Outlet-2, "description": "on"},
+                         outletStatus_2: {"status": 0, "name": Outlet-3, "description": "off"},
+                         outletStatus_3: {"status": 0, "name": Outlet-4, "description": "off"}
+                        }
+                    }
+                }
         """
         if params is None:
             params = {}
 
-        # Make an initial attempt at connection.
-        # Allows us to fail early if misconfigured.
-        yield self.ibootbar.run_snmp_get(session)
-        if not self.ibootbar.oid_cache['ibootbar_connection'].get('connected', False):
-            self.log.error('No initial SNMP response.')
-            self.log.error('Either there is a network connection issue, '
-                           + 'or maybe you are using the wrong SNMP '
-                           + 'version. Either way, we are exiting.')
-
-            reactor.callFromThread(reactor.stop)
-            return False, 'acq process failed - No connection to ibootbar'
+        # Set initial default outlet names
+        names = ['Outlet-1', 'Outlet-2', 'Outlet-3', 'Outlet-4',
+                     'Outlet-5', 'Outlet-6', 'Outlet-7', 'Outlet-8']
 
         self.is_streaming = True
-
         while self.is_streaming:
-            yield self.ibootbar.run_snmp_get(session)
-            self.log.debug("{data}", data=session.data)
             yield dsleep(1)
+            read_time = time.time()
+
+            # Check if 60 seconds has passed before getting status
+            if (read_time - self.lastGet) < 60:
+                continue
+
+            get_list = []
+            name_list = []
+            names = []
+
+            # Create the lists of OIDs to send get commands
+            for i in range(8):
+                get_list.append(('IBOOTPDU-MIB', 'outletStatus', i))
+                name_list.append(('IBOOTPDU-MIB', 'outletName', i))
+
+            # Issue SNMP GET command
+            get_result = yield self.snmp.get(get_list, self.version)
+            name_result = yield self.snmp.get(name_list, self.version)
+
+            for item in name_result:
+                names.append(item[1].prettyPrint())
+            
+            # Do not publish if ibootbar connection has dropped
+            try:
+                message = _build_message(get_result, names, read_time)
+                self.log.debug("{msg}", msg=message)
+                session.app.publish_to_feed('ibootbar', message)
+            except ConnectionError as e:
+                self.log.error(f'{e}')
```

Comment:
I think I'm missing something here. Where does the ConnectionError come up? I would have thought in the `publish_to_feed` call, but that would be if the crossbar connection has dropped, not if the ibootbar connection dropped, as indicated by the comment. If the ibootbar disconnects, what comes back from the above `snmp.get`'s?

`session.data` is then still updated, even if we hit this, which seems inconsistent.

Reply:
Yep, you're right, moving `update_cache()` inside the `try` block. If the ibootbar disconnects, `snmp.get` returns `None` which then raises a `ConnectionError` in `update_cache()`

Reply:
Ah, yeah, I see where this is coming from when looking at the M1000 agent. The first thing it does there is `update_cache()`, which would hit `ConnectionError` raise like you said: https://github.com/simonsobs/socs/blob/d12fea99596ac553628d551dc005e504dbcb3ee6/agents/meinberg_m1000/meinberg_m1000_agent.py#L298-L311